server/src/test/java/org/elasticsearch/action/admin/cluster/reroute/ClusterRerouteResponseTests.java

```diff
@@ -82,8 +82,10 @@ public void testToXContent() throws IOException {
                     "        \"attributes\" : { }\n" +
                     "      }\n" +
                     "    },\n" +
-                    "    \"metadata\" : {\n" +
+                    "    \"meta-data\" : {\n" +
```

Comment:
If anyone was relying on the stability of the cluster state APIs, this is definitely the biggest thing that would break them. I wonder if we should change `MetaData.toXContent` to have metadata there, and be lenient in the `MetaData.fromXContent` method, allowing both. This means that the internal storage format can handle the change, and the user-facing cluster state will be kept the same as well.

Reply:
An alternative would be to use `meta-data` / `metadata` in `MetaData.toXContent` depending on the context (API vs GATEWAY).

Reply:
Yes I think this makes sense. Will try to parametrize the code.